lib/spack/spack/environment/environment.py

```diff
@@ -883,9 +883,57 @@ def _process_definition(self, item):
             else:
                 self.spec_lists[name] = user_specs
 
+    def _process_view(self, env_view: Optional[Union[bool, str, Dict]]):
+        """Process view option(s), which can be boolean, string, or None.
+
+        A boolean environment view option takes precedence over any that may
+        be included. So ``view: True`` results in the default view only. And
+        ``view: False`` means the environment will have no view.
+
+        Args:
+            env_view: view option provided in the manifest or configuration
+        """
+
+        def add_view(name, values):
+            """Add the view with the name and the string or dict values IF
+            there is not already a view with that name."""
+            if name not in self.views:
```

Comment:
Is this check still doing anything?

Reply:
By that I take it you are asking if it is necessary to check whether the name is in the view.  Good point.  Let me confirm that it is not needed.

Reply:
Well spotted.  It is no longer needed with the environment config/scope changes.